client/components/products.js

```diff
@@ -1,47 +1,203 @@
 import React, {Component} from 'react'
 import {connect} from 'react-redux'
-import {Link} from 'react-router-dom'
+
 import {logout} from '../store'
+import {loadPage} from '../store/divided'
 import {getProducts} from '../store/products'
-
+import {getProduct} from '../store/product'
+import Product from './product'
+import Search from './Search'
+import {getCategories} from '../store/categories'
+import Button from 'react-bootstrap/Button'
+import Card from 'react-bootstrap/Card'
+import Container from 'react-bootstrap/Container'
+import Pagination from 'react-bootstrap/Pagination'
+import {lowToHigh} from '../store/products'
+import {highToLow} from '../store/products'
+import {Categories, aToz, zToa} from '../store/products'
+import queryString from 'query-string'
 //ASSIGNED TO: Aleks
 
 class Products extends Component {
+  constructor(props) {
+    console.log('props', props)
+    console.log('from props', queryString.parse(props.location.search))
+    super()
+  }
+  //
+
+  componentDidUpdate(prevState) {
+    const push = this.props.history.push
+    let sortBy
+
+    if (queryString.parse(this.props.location.search).sortBy) {
+      sortBy = queryString.parse(this.props.location.search).sortBy
+    } else {
+      sortBy = 'AtoZ'
+    }
+    const page = this.props.match.params.page || 1
+    //sortBy === 'undefined' ? 'AtoZ' : sortBy
+    if (prevState.location.search.slice(8) !== sortBy) {
+      push(`/products/${page}?sortBy=${sortBy}`)
+      this.props.load(sortBy, page)
+    }
+  }
+
   componentDidMount() {
-    this.props.load()
+    const sortBy = queryString.parse(this.props.location.search).sortBy
+
+    const page = this.props.match.params.page || 1
+    const push = this.props.history.push
+    this.props.load(sortBy, page)
   }
   render() {
-    const {products} = this.props
-    console.log(products)
+    const {products, divided} = this.props
```

Comment:
create a function outside of the render block and call it inside the render block